src/server/scripts/Spells/spell_quest.cpp

```diff
@@ -1937,6 +1937,97 @@ class spell_q13086_cannons_target : public SpellScriptLoader
         }
 };
 
+enum ThatsAbominable
+{
+    QUEST_THATS_ABOMINABLE                = 13264,
+
+    NPC_ICY_GHOUL                         = 31142,
+    NPC_RISEN_ALLIANCE_SOLDIERS           = 31205,
+    NPC_VICIOUS_GEIST                     = 31147,
+    NPC_RENIMATED_ABOMINATION             = 31692,
+
+    SPELL_ICY_GHOUL_CREDIT                = 59591, // Credit for Icy Ghoul
+    SPELL_VICIOUS_GEISTS_CREDIT           = 60042, // Credit for Vicious Geists
+    SPELL_RISEN_ALLIANCE_SOLDIERS_CREDIT  = 60040, // Credit for Risen Alliance Soldiers
+};
+
+class spell_q13264_thats_abominable : public SpellScriptLoader
+{
+    public:
+        spell_q13264_thats_abominable() : SpellScriptLoader("spell_q13264_thats_abominable") { }
+
+        class spell_q13264_thats_abominable_SpellScript : public SpellScript
+        {
+            PrepareSpellScript(spell_q13264_thats_abominable_SpellScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                return ValidateSpellInfo(
+                {
+                    SPELL_ICY_GHOUL_CREDIT,
+                    SPELL_VICIOUS_GEISTS_CREDIT,
+                    SPELL_RISEN_ALLIANCE_SOLDIERS_CREDIT,
+                });
+            }
+
+            void HandleKnockBack(SpellEffIndex effIndex)
+            {
+                PreventHitDefaultEffect(effIndex);
+
+                if (Creature* creature = GetHitCreature())
+                    if (Unit* charmer = GetCaster()->GetCharmerOrOwner())
+                        if (Player* player = charmer->ToPlayer())
+                            if (player->GetQuestStatus(QUEST_THATS_ABOMINABLE) == QUEST_STATUS_INCOMPLETE)
+                                if (GiveCreditIfValid(player, creature))
+                                    creature->KillSelf();
+            }
+
+            bool GiveCreditIfValid(Player* player, Creature* creature)
+            {
+                uint32 entry = creature->GetEntry();
+
+                switch(entry)
+                {
+                    case NPC_ICY_GHOUL:
+                        player->CastSpell(player, SPELL_ICY_GHOUL_CREDIT, true);
+                        return true;
+                        break;
```

Comment:
unreachable code, remove this line

Reply:
What?

Reply:
By removing these line's, player won't receive any quest credit's.

Reply:
`return true;` skips the `break;` line so it becomes unreachable
(either `return` or `break`, but not both).

Reply:
a man should read more about return statement in C++, because it moves you out of the function frame, so break will never happen